internal/worker/pool.go

```diff
@@ -0,0 +1,78 @@
+package worker
+
+import (
+	"context"
+	"log/slog"
+
+	"github.com/supressionstop/xenking_test_1/internal/config"
+	"github.com/supressionstop/xenking_test_1/internal/usecase"
+)
+
+type Pool struct {
+	workers        []*Worker
+	logger         *slog.Logger
+	FirstSyncChan  chan struct{}
+	workerSyncChan chan struct{}
+	workersErrChan chan error
+	ErrChan        chan error
+}
+
+func NewPool(cfg config.Config, logger *slog.Logger, fetchLine usecase.FetchLine) *Pool {
+	pool := &Pool{
+		logger:         logger,
+		workers:        make([]*Worker, len(cfg.Workers)),
+		FirstSyncChan:  make(chan struct{}),
+		workerSyncChan: make(chan struct{}, len(cfg.Workers)),
+		workersErrChan: make(chan error, len(cfg.Workers)),
```

Comment:
It seems to me like an overcomplicated implementation of the worker's pool. I believe we can get rid of `FirstSyncChan`, `ErrChan`, `workersErrChan`, and `workersSyncChan`. We can log workers errors inside worker and remove a worker sync.

Reply:
TODO: try to replace many structs to worker pool pattern.

Reply:
Need to discuss new solution